vllm/engine/async_llm_engine.py

```diff
@@ -217,10 +217,16 @@ async def step_async(self) -> List[RequestOutput]:
         else:
             output = []
 
-        return self._process_model_outputs(
+        request_outputs = self._process_model_outputs(
             output, scheduler_outputs.scheduled_seq_groups,
             scheduler_outputs.ignored_seq_groups)
 
+        # Log stats.
+        if self.log_stats:
+            self.stat_logger.log(self._get_stats(scheduler_outputs))
```

Comment:
Would it be worth considering moving the call to `stat_logger.log()` into `_process_model_outputs()`?
Currently, it's being invoked in both `LLMEngine.step()` and `AsyncLLMEngine.step_async()`, duplicating the functionality.
https://github.com/vllm-project/vllm/blob/fe3b5bbc23a99533bc7d4a94ae073828ed025974/vllm/engine/llm_engine.py#L542-L544
What are your thoughts?

Reply:
feel like having it here makes more sense for abstraction? it is technically not related to process model output

Reply:
+1, i think we can just fold it into `_process_model_outputs`, maybe make `_process_model_outputs` accept `scheduler_outputs` instead of `scheduled_seq_groups` and    `ignored_seq_groups`?

Reply:
Hmm I don' t have very strong opinion here, but I think relevant logging should go to the relevant API. E.g., logging about e2e stats is a little bit weird to be in process model output API, but it makes sense to be in step API. It is fine now because of all assuptions we have (process_model_output is called only by step), so not very strong at the moment